R/check_functions.R

```diff
@@ -25,3 +25,10 @@ check_vars_in_vector <- function(vec, vars, vec_nam) {
                                   paste(other_vars, collapse = ", "))
   return()
 }
+
+check_logicalArgument <- function(arg, argName) {
+  if(!is.logical(arg) || length(arg) != 1) {
+    stop("'", argName, "' needs to be a logical vector of length 1.")
+  }
+  return()
+}
```

Comment:
Nit: Maybe something like this (checked it only superficially - I don't know whether `match.fun()` is robust enough):

```r
check_typeArgument <- function(arg, argName, type = c("logical", "numeric", "character")) {
  type <- match.arg(type)
  check_fun <- match.fun(paste0("is.", type))

  if(!check_fun(arg) || length(arg) != 1) {
    stop("'", argName, "' needs to be a ", type, " vector of length 1.")
  }
  return()
}
check_typeArgument("hello", "hello", "character")
check_typeArgument("hello", "hello", "numeric")
check_typeArgument("hello", "hello", "logical")
```

Reply:
I see the point of reducing code duplication, but I think 
a) this decreases code readibility and 
b) it is sometimes easier to have functions have different names (`is.numeric`, `is.character` etc.) than to change behavior via arguments (`is.vectortype(type = "numeric)` etc,).
Still sth. we could discuss!

Reply:
I totally agree. I also thought that the code would become too abstract.